### ISSUE TITLE
Unable to use ClassMemoryLoss to train the model

### ISSUE THREAD

**Issue by ArminLee** [ShengcaiLiao/QAConv#23]
In the QAConv codes, I tried to modify the loss function to ClassMemoryLoss as the criterion but the acc is nearly zero. Is the ClassMemoryLoss available to use? Are ClassMemoryLoss and Focal Loss in the paper the same? The code is shown below.

> criterion = ClassMemoryLoss(matcher, num_classes, num_features, hei, wid).cuda()

**Comment by ShengcaiLiao** [comment]
Hi, thanks for the feedback.

Yes, the ClassMemoryLoss and the Focal Loss inside are the same in the paper. They should still be useful. But indeed, the acc at the beginning is quite low. How many epochs did you run in observing the zero acc?

**Comment by ArminLee** [comment]
I have trained for about 20 epochs and the acc is still below 1%. Also, the acc fluctuate from 0% to 1%. The log of training is attached below.

![image](https://user-images.githubusercontent.com/45883628/154041008-2ed65e12-f6bd-4257-93e3-03f85f9b1f08.png)

**Comment by ShengcaiLiao** [comment]
Thanks for the log. The class memory loss is not stable, and in some rare situations it cannot converge. In such cases, please try several runs.

Besides, it is slow in large-scale learning. 

That's why I have been working on improving the code and finally decided to remove the class memory loss. Please refer to the [change log](https://github.com/ShengcaiLiao/QAConv/blob/master/CHANGELOG.md). For example, in QAConv 1.1 a pre-training function is included to stabilize the training. If you do need the class memory loss, QAConv 1.2 would be recommended, which is the last version supporting class memory loss. The [TransMatcher](https://github.com/ShengcaiLiao/QAConv/tree/master/projects/transmatcher) code can also be tried, which is built upon QAConv 1.2 and still use the class memory loss.

**Comment by ArminLee** [comment]
Thanks for your reply! I will try to use it in the previous version.
